src/serde_macros.rs

```diff
@@ -64,11 +64,9 @@ macro_rules! serde_impl(
                             E: ::serde::de::Error,
                         {
                             if v.len() != $t::LEN {
-                                Err(E::invalid_length(v.len(), &stringify!($len)))
+                                Err(E::invalid_length(v.len(), &format!("{}", $t::LEN).as_str()))
                             } else {
-                                let mut ret = [0; $len];
-                                ret.copy_from_slice(v);
-                                Ok($t(ret))
+                                Ok($t::from_slice(v).expect("length already checked"))
```

Comment:
At least nominally this adds an extra error path

Reply:
Yeah this is not related to removing the `$len` matcher, but because the newtypes have a `Hash` as inner type and not a `[u8; $len]`.

One thing I thought about is to `map_err` the `from_slice` error into the serde error which would avoid a panic in the scenario where `from_slice` adds another error possibility.

Reply:
I changed it to use map_err, to at least exclude a panic possibility.